### ISSUE TITLE
sql+cli: cannot dump UUID columns

### ISSUE THREAD

**Issue by jtescher** [cockroachdb/cockroach#18501]
SQL Dump does not seem to support UUID column types.

Steps to reproduce:

```
CREATE DATABASE test;
CREATE TABLE test.uuid (id UUID NOT NULL);
INSERT INTO test.uuid values ('e9716c74-2638-443d-90ed-ffde7bea7d1d');
```

Then running `./cockroach dump test --insecure` results in the following error:

```
# Server version: CockroachDB CCL v1.1-beta.20170907 (linux amd64, built 2017/09/13 03:03:08, go1.8.3) (same version as client)
# Cluster ID: 4628dcf5-e6e9-47a7-9b54-aef396723c9f
CREATE TABLE "uuid" (
	id UUID NOT NULL,
	FAMILY "primary" (id, rowid)
);
E170915 00:28:38.448152 1 util/log/crash_reporting.go:114  a panic has occurred!
panic: unknown []byte type: e9716c74-2638-443d-90ed-ffde7bea7d1d, id: UUID [recovered]
	panic: unknown []byte type: e9716c74-2638-443d-90ed-ffde7bea7d1d, id: UUID

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.RecoverAndReportPanic(0x7f9412274438, 0xc420014460, 0xc420442000)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:91 +0x9f
panic(0x1aed700, 0xc42026ce40)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/cli.dumpTableData(0x2a305e0, 0xc42000e018, 0xc4206f6700, 0xc42026c460, 0x1e, 0x3b, 0xc4206fe400, 0x1, 0x1caa966, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/dump.go:473 +0x123c
github.com/cockroachdb/cockroach/pkg/cli.runDump(0x2a19c80, 0xc42025f7a0, 0x1, 0x2, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/dump.go:113 +0xa3a
github.com/cockroachdb/cockroach/pkg/cli.MaybeDecorateGRPCError.func1(0x2a19c80, 0xc42025f7a0, 0x1, 0x2, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:35 +0x5a
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).execute(0x2a19c80, 0xc42025f6c0, 0x2, 0x2, 0x2a19c80, 0xc42025f6c0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:644 +0x3f8
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2a17ac0, 0x6c0f9e, 0xc4202ab750, 0xc4206a1f10)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:734 +0x339
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).Execute(0x2a17ac0, 0xc4206a1f00, 0xc4202ab720)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:693 +0x2b
github.com/cockroachdb/cockroach/pkg/cli.Run(0xc4200101d0, 0x3, 0x3, 0xc420014460, 0xc420442000)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:141 +0x6d
github.com/cockroachdb/cockroach/pkg/cli.Main()
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:50 +0x18d
main.main()
	/go/src/github.com/cockroachdb/cockroach/main.go:26 +0x20
```

Nothing special in the logs:

```
I170915 00:37:32.773638 363 sql/executor.go:715  [client=127.0.0.1:38074,user=root,n1] execRequest: SELECT rowid, id FROM test."uuid" AS OF SYSTEM TIME '1505435852759599189.0000000000' ORDER BY PRIMARY KEY test."uuid" LIMIT 10000
I170915 00:37:32.773752 363 sql/executor.go:1038  [client=127.0.0.1:38074,user=root,n1] executing 1/1: SELECT rowid, id FROM test."uuid" AS OF SYSTEM TIME '1505435852759599189.0000000000' ORDER BY PRIMARY KEY test."uuid" LIMIT 10000
I170915 00:37:32.773767 363 sql/executor.go:1353  [client=127.0.0.1:38074,user=root,n1] SELECT rowid, id FROM test."uuid" AS OF SYSTEM TIME '1505435852759599189.0000000000' ORDER BY PRIMARY KEY test."uuid" LIMIT 10000
I170915 00:37:32.775483 363 sql/executor.go:1852  [client=127.0.0.1:38074,user=root,n1] not distributing query
I170915 00:37:32.776033 363 sql/executor.go:1353  [client=127.0.0.1:38074,user=root,n1] SELECT (1 results) done
```

**Comment by knz** [comment]
@lego were you not looking at this recently?

**Comment by lego** [comment]
Not this in particular but I'll make the PR for this fix right now along with all the other changes to UUID I had made.

**Comment by lego** [comment]
Thanks for reporting this, it should be fixed now on master and will be slated for 1.1 :)
